Updated docs to include git submodule instructions

madhavajay opened the pull request:
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
- Improved setup instructions

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Did you run `make test-all`?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] New Unit tests added
- [ ] Unit tests pass locally with my changes

madhavajay committed:
Updated docs to include git submodule instructions

chinmayshah99 commented on contributing.md:
````diff
@@ -11,6 +11,16 @@ Regarding Pybind, Here is a good [introduction](https://www.youtube.com/watch?v=
+Currently we are using git submodules to allow editors like VSCode to locate source code, this should be refactored later. Don't forget to init and update the submodules or you will get an error on building.
+
+```
+$ git submodule update --init --recursive
````
We already have the script for same at https://github.com/OpenMined/PyDP/blob/dev/ext_source_setup. 
Is this complexity needed?

madhavajay replied:
Hey, yeah good point sorry, I didn't actually see that, but perhaps the README.md wasn't clear to me, I have fixed a few issues and updated the README to explain the three different shell scripts and cleaned up some consistency issues etc. Not trying to add unnecessary noise to the git history so feel free to reject, but I did miss the submodule thing initially and wasted a bit of time debugging bazel errors.

madhavajay committed:
Updated docs to explain all three shell scripts

- Renamed shell scripts to .sh extension for consistency
- Added +x execution bit to shell scripts

madhavajay committed:
Fixed broken CI tests
